linkml_runtime/utils/schemaview.py

```diff
@@ -107,21 +113,28 @@ def detect_cycles(f: Callable[[Any], Iterable[Any] | None], x: Any) -> None:
     GREY:  node is being processed; processing includes exploring all neighbours reachable via f(node)
     BLACK: node and all of its neighbours (and their neighbours, etc.) have been processed
 
-    A directed cycle reachable from node `x` raises a ValueError.
+    A directed cycle reachable from a node or its neighbours raises a ValueError.
 
     :param f: function that returns an iterable of neighbouring nodes (parents or children)
     :type f: Callable[[Any], Iterable[Any] | None]
-    :param x: graph node
-    :type x: Any
-    :raises ValueError: if a cycle is discovered through repeated calls to f(x)
+    :param node_list: list or other iterable of values to process
+    :type node_list: Iterable[Any]
+    :raises ValueError: if a cycle is discovered through repeated calls to f(node)
     """
+    # ensure we have some nodes to start the analysis
+    if not node_list or not isinstance(node_list, Iterable) or isinstance(node_list, str):
```

Comment:
make sure that the input is a non-empty iterable and that it isn't a string (which is an iterable, but probably not one that we need to detect cycles in!).